### PR TITLE
fix: Use encryption for SMS confidential parameters [DHIS2-10076]

### PR DESCRIPTION
**Pull request by zubaira**

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/sms/config/BulkSmsHttpGateway.java [hunk 1]
```diff
@@ -87,7 +96,8 @@ public OutboundMessageResponse send( String subject, String text, Set<String> re
 
     private HttpHeaders getRequestHeaderParameters( BulkSmsGatewayConfig bulkSmsGatewayConfig )
     {
-        String credentials = bulkSmsGatewayConfig.getUsername().trim() + ":" + bulkSmsGatewayConfig.getPassword().trim();
+        String credentials = bulkSmsGatewayConfig.getUsername().trim() + ":" +
+            pbeStringEncryptor.decrypt( bulkSmsGatewayConfig.getPassword().trim());
```
Comment (stian-sandvold) [inline dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/sms/config/BulkSmsHttpGateway.java:100]:
```suggestion
            pbeStringEncryptor.decrypt( bulkSmsGatewayConfig.getPassword().trim() );
```

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/sms/config/SimplisticHttpGetGateWay.java [hunk 2]
```diff
@@ -184,8 +193,8 @@ public OutboundMessageResponse send( String subject, String text, Set<String> re
         {
             if ( !parameter.isHeader() )
             {
-                valueStore.put( parameter.getKey(), parameter.getValue() );
-            }
+                valueStore.put( parameter.getKey(), parameter.isConfidential() ?
+                    pbeStringEncryptor.decrypt( parameter.getValue() ) : parameter.getValue() );            }
```
Comment (stian-sandvold) [inline dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/sms/config/SimplisticHttpGetGateWay.java:197]: closing bracket on wrong line :)

**Review by stian-sandvold — APPROVED** [review]

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/sms/config/GenericGatewayParameter.java [hunk 3]
```diff
@@ -106,4 +107,30 @@ public void setEncode( boolean encode )
     {
         this.encode = encode;
     }
+
+    @Override
+    public boolean equals( Object o )
```
Comment (gnespolino) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/sms/config/GenericGatewayParameter.java:112]: Can't we just annotate this class with:
```java
@EqualsAndHashCode(of = {"key","value", "confidential", "encode", "header"})
```
?

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/sms/config/BulkSmsHttpGateway.java [hunk 4]
```diff
@@ -50,6 +52,13 @@
 public class BulkSmsHttpGateway
```
Comment (gnespolino) [inline dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/sms/config/BulkSmsHttpGateway.java:52]:
```java
@RequiredArgsConstructor
```
with
```java
@Qualifier( "tripleDesStringEncryptor" )
private final PBEStringEncryptor pbeStringEncryptor;
```

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/sms/config/ClickatellHttpGateway.java [hunk 5]
```diff
@@ -50,6 +52,13 @@
 public class ClickatellHttpGateway
```
Comment (gnespolino) [inline dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/sms/config/ClickatellHttpGateway.java:52]: same as `BulkSmsHttpGateway`

**Review by gnespolino — APPROVED** [review]
optional minor style improvement suggestions

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/sms/config/SimplisticHttpGetGateWay.java (hunk 2 above)
Comment (zubaira) [reply dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/sms/config/SimplisticHttpGetGateWay.java]: resolved

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/sms/config/GenericGatewayParameter.java (hunk 3 above)
Comment (zubaira) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/sms/config/GenericGatewayParameter.java]: resolved

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/sms/config/BulkSmsHttpGateway.java (hunk 4 above)
Comment (zubaira) [reply dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/sms/config/BulkSmsHttpGateway.java:54]: resolved

**Pull request merged**